### PR TITLE
Drop invalid asserts

### LINKED ISSUE #32856: Assertion 'unit' failed at src/journal/journald-server.c:3015, function config_parse_compress(). Aborting.
### systemd version the issue has been seen with

256-rc2

### Used distribution

Debian Unstable

### Linux kernel version used

6.7.12-amd64

### CPU architectures issue was seen on

x86_64

### Component

journalctl

### Expected behaviour you didn't see

System fails to boot to Xorg after updating to 256-rc2 with the following message in dmesg:

systemd-journald[468]: Assertion 'unit' failed at src/journal/journald-server.c:3015, function config_parse_compress(). Aborting.

Reverting back to 255 fixes the problem.

I've reported this to Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1071201

### Unexpected behaviour you saw

_No response_

### Steps to reproduce the problem

_No response_

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by dtardon**
Fixes #32856

### CODE REVIEW

File: test/units/TEST-74-AUX-UTILS.pstore.sh [hunk 1]
```diff
@@ -64,13 +64,13 @@ prepare_efi_logs() {
...
     while read -r line; do
-        filename="$(printf "dmesg-efi-%0.10d%0.2d%0.3d" "$timestamp" "$chunk" "$count")"
+        filename="$(printf "dmesg-efi_pstore-%0.10d%0.2d%0.3d" "$timestamp" "$chunk" "$count")"
         echo "$line" >"/sys/fs/pstore/$filename"
         chunk=$((chunk + 1))
     done < <(sed '1!G;h;$!d' "$file")
 
     if [[ "$chunk" -eq 0 ]]; then
-        echo >&2 "No dmesg-efi files were created"
+        echo >&2 "No dmesg-efi_pstore files were created"
```
Comment (yuwata) [inline test/units/TEST-74-AUX-UTILS.pstore.sh]: What? Is this related? Sneaked from somewhere?

Comment (dtardon) [reply test/units/TEST-74-AUX-UTILS.pstore.sh]: Oops, sorry.

**Review by bluca — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

**Comment by obfusk** [comment]
> config_parse_compress() is not used for parsing unit files, hence unit is always NULL.

Maybe the tests in `src/journal/test-journald-config.c` should also call functions like `config_parse_compress()` with `NULL` instead of `""`? Presumably that would have caught this regression.
